components/core/src/clp_s/search/AddTimestampConditions.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef CLP_S_ADD_TIMESTAMP_CONDITIONS_HPP
+#define CLP_S_ADD_TIMESTAMP_CONDITIONS_HPP
+
+#include <optional>
+
+#include "../Defs.hpp"
+#include "../TimestampDictionaryReader.hpp"
+#include "Transformation.hpp"
+
+namespace clp_s::search {
+class AddTimestampConditions : public Transformation {
+public:
+    // Constructors
+    AddTimestampConditions(
+            std::shared_ptr<TimestampDictionaryReader> const& timestamp_dict,
+            std::optional<epochtime_t> begin_ts,
+            std::optional<epochtime_t> end_ts
+    )
+            : m_timestamp_dict(timestamp_dict),
```

Comment:
* If we do keep the computation of the authoritative timestamp column name inside this class, would it be more appropriate to do it here?
* Should we really allow both timestamps to be empty?

Reply:
Per other comment let's move it outside of the class.

And yeah I prefer letting both timestamps be empty to minimize responsibilities for the caller. The OrOfAndForm normalization pass doesn't have any problems unwrapping the AndExpr(Expr) that results if the begin and end timestamp are both missing, so aren't any real problems if this happens.